### PR TITLE
CLOUDP-79635: Get a private endpoint from an Atlas project

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-79635

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

**AWS**
```
./bin/mongocli atlas privateendpoint aws interface describe 600994a349b14c5d7b83098f --privateEndpointId vpce-0df793f9c89c24a3b -o json

{
  "interfaceEndpointId": "vpce-0df793f9c89c24a3b",
  "deleteRequested": false,
  "connectionStatus": "AVAILABLE"
}

./bin/mongocli atlas privateendpoint aws interface describe 600994a349b14c5d7b83098f --privateEndpointId vpce-0df793f9c89c24a3b

ID                       STATUS      ERROR
vpce-0df793f9c89c24a3b   AVAILABLE

```
**Azure**
```
./bin/mongocli atlas privateendpoint azure interface describe 6009a7a62cde816e1914b656 --privateEndpointId %2Fsubscriptions%2F19265c27-b60e-4c3b-9426-ae3f507300b5%2FresourceGroups%2Ftest%2Fproviders%2FMicrosoft.Network%2FprivateEndpoints%2Ftest -o json

{
  "privateEndpointConnectionName": "19265c27-b60e-4c3b-9426-ae3f507300b5/test/test",
  "privateEndpointIPAddress": "10.0.0.4",
  "privateEndpointResourceId": "/subscriptions/19265c27-b60e-4c3b-9426-ae3f507300b5/resourceGroups/test/providers/Microsoft.Network/privateEndpoints/test",
  "deleteRequested": false
}

 ./bin/mongocli atlas privateendpoint azure interface describe 6009a7a62cde816e1914b656 --privateEndpointId %2Fsubscriptions%2F19265c27-b60e-4c3b-9426-ae3f507300b5%2FresourceGroups%2Ftest%2Fproviders%2FMicrosoft.Network%2FprivateEndpoints%2Ftest

ID                                                                                                                          IP ADDRESS   STATUS   ERROR
/subscriptions/19265c27-b60e-4c3b-9426-ae3f507300b5/resourceGroups/test/providers/Microsoft.Network/privateEndpoints/test   10.0.0.4
```

### CODE REVIEW

File: internal/mocks/mock_default_mongodb_version.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
+// Code generated by MockGen. DO NOT EDIT.
+// Source: github.com/mongodb/mongocli/internal/store (interfaces: DefaultVersionGetter)
+
+// Package mocks is a generated GoMock package.
+package mocks
```
Comment (andreaangiolillo) [inline internal/mocks/mock_default_mongodb_version.go:5]: I am not sure why I have this mock file to add

**Comment by gssbzn** [comment]
[q] before I go deep o technical side, I see to escape the slashes for azure, is this because of bash or is the client? if bash, can you do `"/subscriptions/19265c27-b60e-4c3b-9426-ae3f507300b5/resourceGroups/test/providers/Microsoft.Network/privateEndpoints/test"` with quotes to avoid the escape? if not we may need to submit a PR to the client to fix the escaping there

File: internal/mocks/mock_default_mongodb_version.go (hunk 1 above)
Comment (gssbzn) [reply internal/mocks/mock_default_mongodb_version.go:5]: I may have forgotten to generate mocks when added that method

**Review by gssbzn — APPROVED** [review]
LGTM but let's make sure where the need to use escape characters is coming from as this may be troublesome for some users if we don't document it well

**Comment by andreaangiolillo** [comment]
no, using double-quote doesn’t work I got

```
./bin/mongocli atlas privateendpoint azure interface describe 6009a7a62cde816e1914b656 --privateEndpointId "/subscriptions/19265c27-b60e-4c3b-9426-ae3f507300b5/resourceGroups/test/providers/Microsoft.Network/privateEndpoints/test"

Error: GET https://cloud-dev.mongodb.com/api/atlas/v1.0/groups/5e4e593f70dfbf1010295836/privateEndpoint/AZURE/endpointService/6009a7a62cde816e1914b656/endpoint//subscriptions/19265c27-b60e-4c3b-9426-ae3f507300b5/resourceGroups/test/providers/Microsoft.Network/privateEndpoints/test: 404 (request "RESOURCE_NOT_FOUND") Cannot find resource /api/atlas/v1.0/groups/5e4e593f70dfbf1010295836/privateEndpoint/AZURE/endpointService/6009a7a62cde816e1914b656/endpoint//subscriptions/19265c27-b60e-4c3b-9426-ae3f507300b5/resourceGroups/test/providers/Microsoft.Network/privateEndpoints/test.
```

The [documentation](https://docs.atlas.mongodb.com/reference/api/private-endpoints-endpoint-get-one) suggests using this format for the path param`ENDPOINT-ID`

Screenshot 2021-01-25 at 10 41 02

**Comment by gssbzn** [comment]
We'll address the escape issue in the go client, we have opened a ticket for it

**Pull request merged**